### PR TITLE
fix(login): rejig for better validations & typos

### PR DESCRIPTION
**Pull request by raisedadead**
This change should address some of the typos and does slightly better validations than before. Anyways, the PR is doing a lot, so breaking up the user stories for the review. 

Broadly this PR is:

- [x] Updating to a better email validation logic based on a SO answer.
- [x] Using a lightweight wrapper to check for common typos like `gmal` vs. `gmail` (works with many common domain names like Yahoo, Hotmail, etc.) 
- [x] The email submission is mainly non-intrusive. If you enter the email correctly the first time, you will see no errors or warnings, but you will be asked to confirm if you made a known typo.
- [x] Moving around some third-party libs to our CDN (for reliability) & dropping IE support.
- [x] Show the email field in read-only mode at all times. This should give the user even more clarity when they enter a 'valid' email, but we couldn't suggest any typos: ex: `rob@gmail.com` vs. `bob@gmail.com`

### CODE REVIEW

**Comment by RandellDawson** [comment]
Where is the jQuery imported?  I see jQuery syntax, but did not quickly see a CDN for it in this file?

If possible, I think rewriting to not use jQuery would help speed up the page a bit.

**Comment by raisedadead** [comment]
Correct, we do not use jQuery. 

Instead, we have been using a lightweight alternative called '[cash](https://github.com/fabiospampinato/cash)' for a while. It gives the same mental model & syntax at a much cheaper cost.

Yeah - it is possible to move away, but I will look into it later. Right now I am trying to address some email bouncing issues because the validation is a bit weak and people are making typos all the time.

**Comment by raisedadead** [comment]
These changes are now live on the staging site for you to test.

File: pages/login.html [hunk 1]
```diff
@@ -542,1078 +569,201 @@ <h3 class="text">
...
-      function passwordlessSendEmail() {
-        
-        // Reset any error messages
-        removeErrorMessage();
-        
-        var email = $('#email').val();
+      // Email validation helpers
+      const isEmail = (email) =>
+        email.match(
+          /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
Comment (ojeytonwilliams) [inline pages/login.html:631]: I'm going to have to take your word for this one!

File: pages/login.html [hunk 2]
```diff
@@ -460,6 +476,13 @@ <h3 class="separator"><span>or</span></h3>
                     <button class="btn btn-primary" id="send-email-btn">
                       Email me a sign in code
                     </button>
+                    <button
+                      class="btn btn-primary"
+                      id="confirm-email-btn"
+                      style="display: none"
+                    >
+                      Confirm & Email me a sign in code
```
Comment (ojeytonwilliams) [inline pages/login.html:484]:
```suggestion
                      Confirm & email me a sign in code
```
nit

**Review by ojeytonwilliams — COMMENTED** [review]
How about validating onblur?  It's slightly friendlier than waiting until the user tries to submit.

File: pages/login.html (hunk 1 above)
Comment (raisedadead) [reply pages/login.html:631]: Shh.. this is from StackOverflow.

**Comment by raisedadead** [comment]
Demo

https://user-images.githubusercontent.com/1884376/201648708-4c95ec00-43c5-4917-a7ce-108765d9fc2b.mp4

**Comment by ojeytonwilliams** [comment]
Hey @raisedadead, this seems to work as intended. Did you want to validate on blur and, if so, do that in this PR?  If the answer to either is 'no', I'll crack on and finish my review.

**Comment by raisedadead** [comment]
> Hey @raisedadead, this seems to work as intended. Did you want to validate on blur and, if so, do that in this PR?  If the answer to either is 'no', I'll crack on and finish my review.

I am pretty sure it is not required because the typo hint and the other validation happen when the user clicks the button.

**Comment by ojeytonwilliams** [comment]
No, it's not required. It's probably better UX, though.

Anyways, if we do that, let's do it separately.

**Review by ojeytonwilliams — APPROVED** [review]
Repeatedly calling `setupAuth0FromConfig` seems a little odd (it feels like it should be a one-time thing), but doesn't seem to be broken.

Which is to say, this LGTM :+1:

**Comment by raisedadead** [comment]
Thanks. 

That is a good observation - but if you look at how it plays out in the context, the user is not spending too long clicking on different options.

So effectively, the setup is done only once.

Also, there is one more reason for this approach. I am working on it (different PR). We will be doing better session refreshing and could be sending links (user preference based) instead of OTP with a slightly different 'setup'.

Anyways do not want to get ahead of myself.

**Pull request merged**